### ISSUE TITLE
Cannot use features from non-enabled language CXX

### ISSUE THREAD

**Issue by U-Mallik** [libuvc/libuvc#215]
When I enable the -DBUILD_TEST switch (-DBUILD_TEST=ON) I receive the following message

Cannot use features from non-enabled language CXX
CMake Generate step failed.  Build files cannot be regenerated correctly.

I have tried all sorts of things like adding CXX to languages and the other things shown in the error cited in  Build test failed #180. To no avail. I also tried to add 

  target_compile_options(uvc_test
    PRIVATE $<$<COMPILE_LANGUAGE:C,CXX>:-std=c++11 -x c++>
  )

and 

  target_compile_options(uvc_test
    PRIVATE -std=c++11 -x c++
  ) 

I use CMAKE version 3.18.4

**Comment by U-Mallik** [comment]
When I do all the above together I receive the NO error from CMAKE. However when I run make I receive the error. I resolve this issue 

um@um-Yoga-A940-27ICB:~/libuvc/build$ make && sudo make install
[ 41%] Built target uvc
[ 45%] Building C object CMakeFiles/uvc_test.dir/src/test.c.o
/home/um/libuvc/src/test.c: In function ‘void cb(uvc_frame_t*, void*)’:
/home/um/libuvc/src/test.c:44:66: error: cast from ‘void*’ to ‘int’ loses precision [-fpermissive]
   44 |   printf("callback! length = %u, ptr = %d\n", frame->data_bytes, (int) ptr);
      |                                                                  ^~~~~~~~~
/home/um/libuvc/src/test.c:44:31: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 2 has type ‘size_t’ {aka ‘long unsigned int’} [-Wformat=]
   44 |   printf("callback! length = %u, ptr = %d\n", frame->data_bytes, (int) ptr);
      |                              ~^               ~~~~~~~~~~~~~~~~~
      |                               |                      |
      |                               unsigned int           size_t {aka long unsigned int}
      |                              %lu
/home/um/libuvc/src/test.c: In function ‘int main(int, char**)’:
/home/um/libuvc/src/test.c:118:52: error: invalid conversion from ‘int’ to ‘void*’ [-fpermissive]
  118 |         res = uvc_start_streaming(devh, &ctrl, cb, 12345, 0);
      |                                                    ^~~~~
      |                                                    |
      |                                                    int
In file included from /home/um/libuvc/src/test.c:37:
/home/um/libuvc/include/libuvc/libuvc.h:622:11: note:   initializing argument 4 of ‘uvc_error_t uvc_start_streaming(uvc_device_handle_t*, uvc_stream_ctrl_t*, void (*)(uvc_frame*, void*), void*, uint8_t)’
  622 |     void *user_ptr,
      |     ~~~~~~^~~~~~~~
/home/um/libuvc/src/test.c:133:13: error: ‘sleep’ was not declared in this scope
  133 |             sleep(1);
      |             ^~~~~
/home/um/libuvc/src/test.c:135:11: error: ‘sleep’ was not declared in this scope
  135 |           sleep(10);
      |           ^~~~~
make[2]: *** [CMakeFiles/uvc_test.dir/build.make:82: CMakeFiles/uvc_test.dir/src/test.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:101: CMakeFiles/uvc_test.dir/all] Error 2
make: *** [Makefile:149: all] Error 2


ERROR FREE CMAKE OUTPUT

um@um-Yoga-A940-27ICB:~/libuvc/build$ cmake .. -DBUILD_TEST=ON -DBUILD_EXAMPLE=ON
-- libusb-1.0 found using pkgconfig
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so (found version "80") 
-- Found JPEG library using standard module
CMake Warning (dev) at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args` (JPEG) does
  not match the name of the calling package (JpegPkg).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindJpegPkg.cmake:58 (find_package_handle_standard_args)
  CMakeLists.txt:48 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so  
-- Building libuvc with JPEG support.
-- Components requested: opencv_core;opencv_highgui
-- Configuring done
-- Generating done
-- Build files have been written to: /home/um/libuvc/build

**Comment by U-Mallik** [comment]
More back ground: When I run "cmake .." only (opencv2 not used) the routines create an executable called "example". This executable successfully opens my camera's UVC device and reports back the following:

um@um-Yoga-A940-27ICB:~/libuvc/build$ ./example
UVC initialized
Device found
Device opened
DEVICE CONFIGURATION (0403:602a/1) ---
Status: idle
VideoControl:
	bcdUVC: 0x0110
VideoStreaming(1):
	bEndpointAddress: 129
	Formats:
	UncompressedFormat(1)
		  bits per pixel: 16
		  GUID: 5955593200001000800000aa00389b71 (YUY2)
		  default frame: 1
		  aspect ratio: 0x0
		  interlace flags: 00
		  copy protect: 00
			FrameDescriptor(1)
			  capabilities: 00
			  size: 640x480
			  bit rate: 16-400000000
			  max frame size: 614400
			  default interval: 1/60
			  interval[0]: 1/60
			FrameDescriptor(2)
			  capabilities: 00
			  size: 1280x720
			  bit rate: 16-400000000
			  max frame size: 1843200
			  default interval: 1/60
			  interval[0]: 1/60
			FrameDescriptor(3)
			  capabilities: 00
			  size: 1920x1080
			  bit rate: 16-400000000
			  max frame size: 4147200
			  default interval: 1/60
			  interval[0]: 1/60
			FrameDescriptor(4)
			  capabilities: 00
			  size: 1280x1024
			  bit rate: 16-400000000
			  max frame size: 2621440
			  default interval: 1/60
			  interval[0]: 1/60
END DEVICE CONFIGURATION

First format: (YUY2) 640x480 60fps
bmHint: 001f
bFormatIndex: 1
bFrameIndex: 1
dwFrameInterval: 166666
wKeyFrameRate: 0
wPFrameRate: 0
wCompQuality: 0
wCompWindowSize: 0
wDelay: 0
dwMaxVideoFrameSize: 614400
dwMaxPayloadTransferSize: 614412
bInterfaceNumber: 1
Streaming...
Enabling auto exposure ...
 ... full AE not supported, trying aperture priority mode
 ... uvc_set_ae_mode failed to enable aperture priority mode: Pipe (-9)
Done streaming.
Device closed
UVC exited

In short the problems begin when I attempt to include opencv2 in the build. Having Opencv2 is imperative because the camera is depending on opencv2 to display its image stream.

**Comment by U-Mallik** [comment]
I was receiving -fpermissive errors  because you convert void *ptr to int using the cast (int) ptr

You have to change this to static_cast<int>(reinterpret_cast<intptr_t>(ptr))

**Comment by U-Mallik** [comment]
I have to delete the LANGUAGES in the project definition. Since C and CXX are automatically initiated as the languages doing this does not cause any anomalous behavior. I received a -fpermissive error because you are casting (void *) ptr to int. Changing it to static_cast<int>(reinterpret_cast<intptr_t>(ptr)) removes the error and allows the compiler to complete its task. My camera is yet to produce a video stream. Therefore, I do not know whether, what I have done is going to have the effect that the original programmers desired.
